server/src/main/java/org/elasticsearch/index/analysis/AnalyzerComponents.java

```diff
@@ -44,6 +44,12 @@ static AnalyzerComponents createComponents(
         final Map<String, CharFilterFactory> charFilters,
         final Map<String, TokenFilterFactory> tokenFilters
     ) {
+        for (String key : analyzerSettings.keySet()) {
+            switch (key) {
+                case "tokenizer", "char_filter", "filter", "type", "position_increment_gap" -> {}
```

Comment:
another way of doing this, without deduplicating the expected analyzer keys, would be to duplicate the settings into a mutable map, which we'd remove from every time we read a certain setting. At the end, if any item is left in the map it means that some unsupported param has been provided.